src/runtime/GlobalObjectBuiltinTypedArray.cpp

```diff
@@ -368,7 +368,7 @@ Value builtinTypedArrayConstructor(ExecutionState& state, Value thisValue, size_
             int numlen = val.toNumber(state);
             int elemlen = val.toLength(state);
             if (numlen != elemlen)
-                ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().TypedArray.string(), false, String::emptyString, errorMessage_GlobalObject_FirstArgumentInvalidLength);
+                ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, state.context()->staticStrings().TypedArray.string(), false, String::emptyString, errorMessage_GlobalObject_FirstArgumentInvalidLength);
```

Comment:
This rewrite is invalid, since the current code follows the ES6 spec. The reason of why the related tc is skipped is that in ES8 the operation has been changed to `toIndex`.